graalpython/com.oracle.graal.python/src/com/oracle/graal/python/builtins/objects/foreign/ForeignObjectBuiltins.java

```diff
@@ -1031,6 +1031,28 @@ protected static long doIt(Object object,
                 gil.acquire();
             }
         }
+        @Specialization(limit = "3", guards="lib.fitsInBigInteger(object)")
+        protected static PInt doBigInteger(Object object,
+                        @Cached PRaiseNode raiseNode,
+                        @CachedLibrary("object") InteropLibrary lib,
+                        @Cached GilNode gil,
+                        @Cached PythonObjectFactory factory) {
+            gil.release(true);
+            try {
+                if (lib.fitsInBigInteger(object)) {
+                    try {
+                        var big = lib.asBigInteger(object);
+                        return factory.createInt(big);
+                    } catch (UnsupportedMessageException e) {
+                        CompilerDirectives.transferToInterpreterAndInvalidate();
+                        throw new IllegalStateException("foreign value claims to be a boolean but isn't");
```

Comment:
```suggestion
                        throw new IllegalStateException("foreign value claims to be a Big Integer but isn't");
```

Reply:
I see this marked as resolved, but no changes to the code were pushed. Is there any reason this suggestion is wrong?

As far as I understand, the current error message is misleading - it says that the "value claims to be boolean" - but it doesn't - we go through the `fitsInBigInteger` check, not `fitsInBoolean` check here.

Reply:
Ah, nevermind. I now see the changes, for some reason I saw the outdated changeset. Sorry.